docs/pages/guides/local-app-development.mdx

```diff
@@ -0,0 +1,79 @@
+---
+title: Local app development
+description: Learn how to compile and build your app locally when using Expo.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+import { BoxLink } from '~/ui/components/BoxLink';
+import { BookOpen02Icon, BuildIcon } from '@expo/styleguide-icons';
+import { CODE } from '~/ui/components/Text';
+
+To build your project into an app locally using your machine, you have to manually generate native code before testing the debug build or creating a production build for it to submit to the app store. There are two ways you can build your app locally. This guide provides a brief introduction to both methods and references to other guides that are necessary to create this workflow.
+
+## Prerequisites
+
+To follow the steps described in this guide, install the following tools on your local machine:
```

Comment:
we might want to link to the relevant sections of react-native docs here for more complete instructions, until we have a chance to write our own (assuming we want to):

- https://reactnative.dev/docs/environment-setup?platform=ios
- https://reactnative.dev/docs/environment-setup?platform=android

Reply:
We do cover how to set up tools in our docs: https://docs.expo.dev/workflow/android-studio-emulator/ & https://docs.expo.dev/workflow/ios-simulator/ but I think if we want to, we can expand on that and also connect this guide and our getting started guide.

Reply:
Added instructions on setting the environment for Android (on macOS, windows, & linux) and iOS (on macOS).

**Preview for Android**
![CleanShot 2023-09-18 at 21 53 21](https://github.com/expo/expo/assets/10234615/76456aba-0288-4404-a100-a1eebbfa975c)

![CleanShot 2023-09-18 at 22 02 25](https://github.com/expo/expo/assets/10234615/f66b8504-9406-469d-8d14-2a299ef2b814)

**Preview for iOS**
![CleanShot 2023-09-18 at 21 52 58](https://github.com/expo/expo/assets/10234615/18d25055-1512-4b3a-a12c-5b60b10cde12)

Reply:
FYI, will follow up with relevant instructions to install Android tools using Studio on Windows and update this guide (https://docs.expo.dev/workflow/android-studio-emulator/) in a follow up PR.